File: exercises/beecrowd/1046/1046.go
// https://judge.beecrowd.com.br/judge/pt/problems/view/1046
package main

import (
	"fmt"
)

func calculatePeriod(beginning, end, collapse int) int {
	duration := (collapse - beginning) + end

	if duration > collapse {
		duration = duration - collapse
	}

	return duration
}

func main() {
	var a, b int

	fmt.Scanf("%d %d", &a, &b)

	fmt.Printf("O JOGO DUROU %d HORA(S)\n", calculatePeriod(a, b, 24))
}

File: exercises/beecrowd/1046/1046.ts
// https://judge.beecrowd.com/pt/problems/view/1046
import * as fs from "fs";

function calculatePeriod(beginning: number, end: number, collapse: number): number {
	let duration = (collapse - beginning) + end

	if (duration > collapse) {
		duration = duration - collapse
	}

	return duration
}

function main(): void {
	const input = fs.readFileSync("/dev/stdin", "utf8");
	const lines = input.split(/\r?\n/);

	const line = String(lines.shift()).split(' ');

	let a = Number(line[0])
	let b = Number(line[1])

	console.log(`O JOGO DUROU ${calculatePeriod(a, b, 24)} HORA(S)`)
}

main();
